aws/cf_template.go

```diff
@@ -325,7 +330,7 @@ func generateTemplate(spec *stackSpec) (string, error) {
 		HealthCheckIntervalSeconds: cloudformation.Ref(parameterTargetGroupHealthCheckIntervalParameter).Integer(),
 		HealthCheckPath:            cloudformation.Ref(parameterTargetGroupHealthCheckPathParameter).String(),
 		HealthCheckPort:            cloudformation.Ref(parameterTargetGroupHealthCheckPortParameter).String(),
-		HealthCheckProtocol:        cloudformation.String("HTTP"),
+		HealthCheckProtocol:        cloudformation.String(protocol),
```

Comment:
I think that's what @mikkeloscar was referring to. We can do that and it will work for new load balancers, but old ones will fail to update because AWS does not support updating the healthcheck protocol of a running LB, maybe it's just for NLBs?

Reply:
No, for ALBs there is not this limitation, there you can change the protocol. Just tested it :)

Reply:
I actually tested this in the real world already, and for my ALB, cloudformation just reconfigured it. It think it replaced the target group with a new one, but the load balancer and listener remained the same.

Reply:
So it's a NLB ~bug~ feature